### DISCUSSION TITLE
how can I use gzip in kong(version:3.4.0)?

### DISCUSSION THREAD

**Discussion by yuchao-java** [Kong/kong#13545] [Help] [answered]
In the Nginx configuration, the gzip module is used to enable and configure compression of the response content to reduce the amount of data transferred, thereby increasing load speed and saving bandwidth. for example:
    gzip  on;
    gzip_types application/javascript text/css image/jpeg image/png image/gif image/jpg;
    gzip_min_length 1000;
    gzip_comp_level 1;

How can I achieve this function quickly and conveniently in kong 3.4.0?

**Comment by chronolaw** [comment]
In Lua you could use `kong.tools.gzip` module.

**Reply by subnetmarco** [reply, marked as answer]
I think he is referring to enabling it on the Kong server as a whole, if we don't provide an easy configuration to configure GZIP on the server using `kong.conf`, then most likely @yuchao-java needs to provide its own Nginx template with those settings enabled: https://docs.konghq.com/gateway/latest/reference/nginx-directives/

**Reply by yuchao-java** [reply]
@subnetmarco Wow, thank you so much, I added the nginx_proxy_gzip configuration item in kong.conf, which achieved my expectation easily!
